x-pack/solutions/security/plugins/security_solution/server/lib/detection_engine/rule_types/esql/utils/check_error_details.ts

```diff
@@ -0,0 +1,26 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+import type { KbnSearchError } from '@kbn/data-plugin/server/search/report_search_error';
+
+const USER_ERRORS_EXCEPTIONS = ['status_exception', 'verification_exception', 'parsing_exception'];
+
+/**
+ * if error can be qualified as user error(configurational), returns isUserError: true
+ * user errors are excluded from SLO dashboards
+ */
+export const checkErrorDetails = (error: unknown): { isUserError: boolean } => {
+  const errorType = (error as KbnSearchError).errBody?.error?.type;
```

Comment:
This assertion seems dangerous;  what if `error` is `undefined` ? I'd maybe add an additional null check to guard against that situation.

Reply:
good catch, thank you